### DISCUSSION TITLE
TLS Secret from Env for Redis List scaler

### DISCUSSION THREAD

**Discussion by normkatz** [kedacore/keda-docs#380] [Q&A]
For [Redis List trigger](https://keda.sh/docs/2.1/scalers/redis-lists/), you have:
enableTLS and I'm assuming the TLS key would be specified here:

authenticationRef:
      name: keda-trigger-auth-redis-secret

For password, you allow passwordFromEnv but I don't think you support redisSecretFromEnv yet.

Can you confirm both of the above and any plans to support Secret from Env?

**Comment by tomkerkhove** [comment]
Good question, maybe @goku321 can elaborate concerning `useTLS` because I'm not sure if you have to specify something

**Reply by norm-seqster** [reply]
To add even more complexity, we have Redis v6 with TLS enabled receiving requests from a Docker container running Ubuntu 18 and phpredis v5.31 driver with these parameters forwarded to redis.so library:
'allow_self_signed' => true,
'ciphers' => 'DEFAULT:!TLSv1.0:!TLSv1.1:!SSLv3',
'local_cert' => '/usr/local/certs/tls.crt',
'local_pk' => '/usr/local/certs/tls.key',
'cafile' => '/etc/ssl/certs/ca.pem'

This is working fine with a self-signed cert in our K8s cluster.  We want to achieve the same connectivity with KEDA.  We need TLS since we're sending PHI which must be encrypted in transit.

**Comment by goku321** [comment]
I acknowledge and will get back on this asap.

**Comment by goku321** [comment]
Hi @norm-seqster , Thanks for your patience. I can confirm that currently redis scalers doesn't support user provided tls key and cert. TLS can be enabled by just setting `enableTLS` to true.

To allow user to provide tls key and cert can be a useful feature to add.
